pkg/cloud/vsphere/services/govmomi/update.go

```diff
@@ -108,19 +108,14 @@ func reconcileNetworkStatus(ctx *context.MachineContext, vm *object.VirtualMachi
 		return errors.Errorf("invalid network count for %q: exp=%d act=%d", ctx, expNetCount, actNetCount)
 	}
 
-	// Update the machine's network status.
-	ctx.Logger.V(4).Info("updating network status")
-	ctx.MachineStatus.Network = allNetStatus
-
 	// Update the MAC addresses in the machine's network config as well. This
 	// is required in order to generate the metadata.
-	for _, netStatus := range allNetStatus {
-		for i := range ctx.MachineConfig.MachineSpec.Network.Devices {
-			dev := &ctx.MachineConfig.MachineSpec.Network.Devices[i]
-			if netStatus.UUID == dev.UUID && netStatus.MACAddr != dev.MACAddr {
-				ctx.Logger.V(4).Info("updating MAC address for device", "device-uuid", dev.UUID, "network-name", dev.NetworkName, "old-mac-addr", dev.MACAddr, "new-mac-addr", netStatus.MACAddr)
-				dev.MACAddr = netStatus.MACAddr
-			}
+	for i := range ctx.MachineConfig.MachineSpec.Network.Devices {
```

Comment:
Wondering if at the very least we should log warning if the two lengths don't match?

Reply:
Hi @andrewsykim,

We log that above. In fact, we return an error.

Reply:
ah, missed it thank you

Reply:
ha, I reported this bug so I should have known this 😆